packages/nuxt/src/app/composables/fetch.ts

```diff
@@ -93,20 +93,18 @@ export function useFetch<
   arg1?: string | UseFetchOptions<_ResT, DataT, PickKeys, DefaultT, ReqT, Method>,
   arg2?: string,
 ) {
-  const [opts = {}, autoKey] = typeof arg1 === 'string' ? [{}, arg1] : [arg1, arg2]
+  const [opts = {}] = typeof arg1 === 'string' ? [{}, arg1] : [arg1, arg2]
 
   const _request = computed(() => toValue(request))
 
-  const _key = computed(() => toValue(opts.key) || hash([autoKey, typeof _request.value === 'string' ? _request.value : '', ...generateOptionSegments(opts)]))
-  if (!_key.value || typeof _key.value !== 'string') {
-    throw new TypeError('[nuxt] [useFetch] key must be a string: ' + _key.value)
+  const key = computed(() => toValue(opts.key) || ('$f' + hash([typeof _request.value === 'string' ? _request.value : '', ...generateOptionSegments(opts)])))
+  if (!key.value || typeof key.value !== 'string') {
```

Comment:
is this still needed ? key should be always ` Ref<string>`

Reply:
probably not!